Modules/FindCUDA/compute_arch.cmake

```diff
@@ -0,0 +1,152 @@
+# Removes duplicates from LIST(s)
+# Usage:
+#   LIST_UNIQUE(<list_variable> [<list_variable>] [...])
+MACRO(LIST_UNIQUE)
+  FOREACH(__lst ${ARGN})
+    IF(${__lst})
+      LIST(REMOVE_DUPLICATES ${__lst})
+    ENDIF()
+  ENDFOREACH()
+ENDMACRO()
+
+################################################################################################
+# A function for automatic detection of GPUs installed  (IF autodetection is enabled)
+# Usage:
+#   DETECT_INSTALLED_GPUS(OUT_VARIABLE)
+FUNCTION(DETECT_INSTALLED_GPUS OUT_VARIABLE)
+  IF(NOT CUDA_GPU_DETECT_OUTPUT)
+    SET(__cufile ${PROJECT_BINARY_DIR}/detect_cuda_archs.cu)
+
+    file(WRITE ${__cufile} ""
+      "#include <cstdio>\n"
+      "int main()\n"
+      "{\n"
+      "  int count = 0;\n"
+      "  if (cudaSuccess != cudaGetDeviceCount(&count)) return -1;\n"
+      "  if (count == 0) return -1;\n"
+      "  for (int device = 0; device < count; ++device)\n"
+      "  {\n"
+      "    cudaDeviceProp prop;\n"
+      "    if (cudaSuccess == cudaGetDeviceProperties(&prop, device))\n"
+      "      std::printf(\"%d.%d \", prop.major, prop.minor);\n"
+      "  }\n"
+      "  return 0;\n"
+      "}\n")
+
+    EXECUTE_PROCESS(COMMAND "${CUDA_NVCC_EXECUTABLE}" "--run" "${__cufile}"
+                    WORKING_DIRECTORY "${PROJECT_BINARY_DIR}/CMakeFiles/"
+                    RESULT_VARIABLE __nvcc_res OUTPUT_VARIABLE __nvcc_out
+                    ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+    IF(__nvcc_res EQUAL 0)
+      STRING(REPLACE "2.1" "2.1(2.0)" __nvcc_out "${__nvcc_out}")
+      SET(CUDA_GPU_DETECT_OUTPUT ${__nvcc_out} CACHE INTERNAL "Returned GPU architetures from detect_gpus tool" FORCE)
+    ENDIF()
+  ENDIF()
+
+  IF(NOT CUDA_GPU_DETECT_OUTPUT)
+    message(STATUS "Automatic GPU detection failed. Building for all known architectures.")
+    SET(${OUT_VARIABLE} ${KNOWN_GPU_ARCHITECTURES} PARENT_SCOPE)
+  ELSE()
+    SET(${OUT_VARIABLE} ${CUDA_GPU_DETECT_OUTPUT} PARENT_SCOPE)
+  ENDIF()
+ENDFUNCTION()
+
+
+################################################################################################
+# Function for selecting GPU arch flags for nvcc based on CUDA_ARCH_NAME
+# Usage:
+#   SELECT_NVCC_ARCH_FLAGS(out_variable)
+FUNCTION(SELECT_NVCC_ARCH_FLAGS out_variable)
+  # List of arch names
+  SET(__archs_names "Fermi" "Kepler" "Maxwell" "All" "Manual")
+  SET(__archs_name_default "All")
+  IF(NOT CMAKE_CROSSCOMPILING)
+    LIST(APPEND __archs_names "Auto")
+    SET(__archs_name_default "Auto")
+  ENDIF()
+
+  # SET CUDA_ARCH_NAME strings (so it will be seen as dropbox in CMake-Gui)
+  SET(CUDA_ARCH_NAME ${__archs_name_default} CACHE STRING "Select target NVIDIA GPU achitecture.")
+  SET_property( CACHE CUDA_ARCH_NAME PROPERTY STRINGS "" ${__archs_names} )
+  mark_as_advanced(CUDA_ARCH_NAME)
+
+  # verIFy CUDA_ARCH_NAME value
+  IF(NOT ";${__archs_names};" MATCHES ";${CUDA_ARCH_NAME};")
```

Comment:
Absolutely - I am a CMake novice so follow safer path to use the same constructs as before :)